src/hotspot/os/linux/attachListener_linux.cpp

```diff
@@ -405,11 +353,23 @@ int LinuxAttachListener::write_fully(int s, char* buf, int len) {
 // if there are operations that involves a very big reply then it the
 // socket could be made non-blocking and a timeout could be used.
 
-void LinuxAttachOperation::complete(jint result, bufferedStream* st) {
-  JavaThread* thread = JavaThread::current();
-  ThreadBlockInVM tbivm(thread);
+void LinuxAttachOperation::effectively_complete_raw(jint result, bufferedStream* st) {
+
+  if (_effectively_completed) {
+    return;
```

Comment:
An assert here would catch lost output that won't be reported anywhere.
```suggestion
    assert(st->size() == 0, "no lost output");
    return;
```

Reply:
A straightforward implementation of this assertion leads to triggering it at restore. 
Seems like `LinuxAttachListener::write_fully()` doesnt reset the buffer_pos in `bufferedStream`, so the `bufferedStream->size() `is non-zero.  

Probably, it worth to add `bufferedStream->reset()` to `LinuxAttachOperation::write_operation_result()`

Reply:
I believe the data there is the text I observe with the CRAllowToSkipCheckpoint. So the root casue should be fixed (the text that won't be shown).

Reply:
Looks like the issue with this assertion was fixed with the commit 29365804b4be9d0cf41a4feaf5ac46fe823d4fba